### PR TITLE
Restructure CurrencyDisplay to use Design System

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

The `CurrencyDisplay` component has legacy CSS and is unfriendly to the new design system.  This is essentially a structural rewrite using the new design system components.

## Screenshots/Screencaps

There should be no visible changes.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by vthomas13 — DISMISSED** [review]

File: ui/components/ui/currency-display/currency-display.component.js [hunk 1]
```diff
@@ -3,6 +3,13 @@ import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import { useCurrencyDisplay } from '../../../hooks/useCurrencyDisplay';
 import { EtherDenomination } from '../../../../shared/constants/common';
+import Box from '../box/box';
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.component.js:6]: Can we try out the new TS version of the `Box`? Component API should be exactly the same so just need to update the import statement 
```suggestion
```

File: ui/components/ui/currency-display/currency-display.component.js [hunk 2]
```diff
@@ -3,6 +3,13 @@ import PropTypes from 'prop-types';
...
+import { Text } from '../../component-library';
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.component.js:7]: Import new TS version of `Box` component 
```suggestion
import { Text, Box } from '../../component-library';
```

File: ui/components/ui/currency-display/currency-display.stories.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { decimalToHex } from '../../../../shared/modules/conversion.utils';
+import CurrencyDisplay from '.';
+
+export default {
+  title: 'Components/UI/CurrencyDisplay',
+
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.stories.js:7]: nit: remove line break
```suggestion
```

File: ui/components/ui/currency-display/currency-display.component.js [hunk 4]
```diff
@@ -3,6 +3,13 @@ import PropTypes from 'prop-types';
...
+import {
+  AlignItems,
+  DISPLAY,
+  FLEX_WRAP,
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.component.js:11]: Can we update these deprecated constants to the new enums? Will need to update instances to use PascalCase
```suggestion
  Display,
  FlexWrap,
```

File: ui/components/ui/currency-display/currency-display.component.js [hunk 5]
```diff
@@ -18,6 +25,10 @@ export default function CurrencyDisplay({
   denomination,
   currency,
   suffix,
+  prefixComponentWrapperProps = {},
+  textProps = {},
+  suffixProps = {},
+  boxProps = {},
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.component.js:31]: Excellent use of inversion of control here by allowing access to these components 💯

File: ui/components/ui/currency-display/currency-display.stories.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      control: {
+        type: 'text',
+      },
+    },
+    denomination: {
+      control: {
+        type: 'text',
+      },
+    },
+  },
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.stories.js:44]: You could update the argTypes to include all props. You will need to import `import { EtherDenomination } from '../../../../shared/constants/common';`

```suggestion
argTypes: {
    className: {
      control: 'text',
    },
    currency: {
      control: 'text',
    },
    'data-testid': {
      control: 'text',
    },
    denomination: {
      control: 'select',
      options: [EtherDenomination.ETH, EtherDenomination.GWEI],
    },
    displayValue: {
      control: 'text',
    },
    hideLabel: {
      control: 'boolean',
    },
    hideTitle: {
      control: 'boolean',
    },
    numberOfDecimals: {
      control: 'number',
    },
    prefix: {
      control: 'text',
    },
    prefixComponent: {
      control: 'text',
    },
    style: {
      control: 'object',
    },
    suffix: {
      control: 'text',
    },
    value: {
      control: 'text',
    },
    prefixComponentWrapperProps: {
      control: 'object',
    },
    textProps: {
      control: 'object',
    },
    suffixProps: {
      control: 'object',
    },
    boxProps: {
      control: 'object',
    },
  },
```

File: ui/components/ui/currency-display/currency-display.stories.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  args: {
+    denomination: 'ETH',
```
Comment (georgewrmarshall) [inline ui/components/ui/currency-display/currency-display.stories.js:46]:
```suggestion
    denomination: EtherDenomination.ETH,
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking great! I like the inversion of control pattern you are implementing here! Less a-prop-calypses and more maintainable components for the win! Made some small suggestions for updating deprecated code and storybook. Checked the component on my local build

Screenshot 2023-06-09 at 11 42 36 AM

**Review by georgewrmarshall — APPROVED** [review]
Code LGTM. Just need pesky e2e tests to pass

**Review by NidhiKJha — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]
Aside from the comment about `boxProps` this looks good to me

File: ui/components/ui/currency-display/currency-display.component.js [hunk 8]
```diff
@@ -28,26 +39,50 @@ export default function CurrencyDisplay({
...
-    <div
+    <Box
       className={classnames('currency-display-component', className)}
       data-testid={dataTestId}
       style={style}
       title={(!hideTitle && title) || null}
+      display={Display.Flex}
+      alignItems={AlignItems.center}
+      flexWrap={FlexWrap.Wrap}
+      {...boxProps}
```
[the remark this answers is not in the document]
Comment (darkwing) [reply ui/components/ui/currency-display/currency-display.component.js]: Good shout, updated!

**Review by garrettbear — APPROVED** [review]

**Review by georgewrmarshall — COMMENTED** [review]
I think the code looks good and I know this PR has been open for a while so this is non-blocking but could we update the story slightly more so it adds more value to the viewer? The default story has 7.3b but it's not reflected in the component. The second story seems to add more value. Can we consolidate them possibly?

Screenshot 2023-07-24 at 3 11 36 PM

**Review by georgewrmarshall — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**